src/libraries/Microsoft.PowerFx.Core/Syntax/Nodes/VariadicBase.cs

```diff
@@ -11,12 +11,13 @@
 namespace Microsoft.PowerFx.Core.Syntax.Nodes
 {
     // Base for all variadic nodes.
-    internal abstract class VariadicBase : TexlNode
+    public abstract class VariadicBase : TexlNode
     {
-        public readonly TexlNode[] Children;
+        // TODO: Would IReadOnlyList<TexlNode> work better here?
+        public TexlNode[] Children { get; }
```

Comment:
Yes, Array is bad because it's mutable. 

IReadOnlyList is the right one (provides both Count and Index)

That reflection test could catch this!